### DISCUSSION TITLE
Synchronous Layout Effect Example

### DISCUSSION THREAD

**Discussion by Jpoliachik** [reactwg/react-native-new-architecture#164] [Q&A] [answered]
I'd like to test out the example [in the docs](https://reactnative.dev/docs/the-new-architecture/landing-page#synchronous-layout-and-effects) using the [sample code linked](https://gist.github.com/lunaleaps/148756563999c83220887757f2e549a3) but `useLayoutEffect` doesn't seem to be properly blocking like expected. 

Steps to reproduce: 
- `npx react-native@latest init SyncTest`
- copied code from [this gist](https://gist.github.com/lunaleaps/148756563999c83220887757f2e549a3) to replace `App.tsx` contents
- `cd ios && bundle install && RCT_NEW_ARCH_ENABLED=1 bundle exec pod install` to enable new architecture for ios. followed [this guide](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-apps.md)
- `yarn ios` 

You'll see the view update and `useLayoutEffect` updates applied afterwards. The `wait(200)` makes this apparent: 
![render-sync-issue-small](https://github.com/reactwg/react-native-new-architecture/assets/5148640/8fd754f6-01b0-4fda-8ab3-2bfc9d7a6bf9)

- I also tried Android with the same result (after enabling new arch in `gradle.properties`)
- I also tried enabling bridgeless mode for iOS and Android following [instructions here](https://github.com/reactwg/react-native-new-architecture/discussions/154)
- I've tried using RN latest 0.73.6 and also 0.74-RC4 with the same result. 
- When metro starts I see logs which indicate fabric and bridgeless should be properly enabled:  `(NOBRIDGE) LOG  Bridgeless mode is enabled \n(NOBRIDGE) LOG  Running "RN0740RC4" with {"rootTag":1,"initialProps":{"concurrentRoot":true},"fabric":true}`

Am I missing any other flags or config required to get synchronous `useLayoutEffect` blocking to work?

**Comment by migueldaipre** [comment]
I believe this hasn't been merged yet

Related link's:

https://github.com/react-native-community/discussions-and-proposals/pull/744

https://github.com/facebook/react-native/pull/43235

**Comment by cortinico** [comment, marked as answer]
It should definitely work if you try on the latest nightly with:

```
npx react-native@latest init RNNightly --version nightly
```

**Comment by Jpoliachik** [comment]
It works on the nightly build! Awesome! Thanks for the help @cortinico

**Comment by DigitalZebra** [comment]
I can confirm this works in RN 75 (and the nightly) when replacing `getBoundingClientRect` with `measureInWindow` and the new architecture is enabled :)
